hivemind/optim/grad_averager.py

```diff
@@ -11,6 +11,10 @@
 logger = get_logger(__name__)
 
 
+TGradientAverager = TypeVar("TGradientAverager", bound="GradientAverager")
+GradientAveragerFactory = Callable[[Type[TGradientAverager], Any], TGradientAverager]
```

Comment:
Question to Max: what's our naming convention for this? In other parts, i've seen Callable[Any, GradientAveragerType].
Personally I'm not sure and fine with either option, but this looks like something that was standardized by more cultured devs :)